script.js

```diff
@@ -0,0 +1,270 @@
+  } else {
+    productdata = JSON.parse(localStorage.getItem("productdata")); // local storage data to object
+  }
+  document.querySelector("#dataTable").innerHTML = "";
+  // if the object has data then it will print it in table format
+  productdata.forEach(function (element, index) {
+    document.querySelector("#dataTable").innerHTML += `<tr class="text-center">
+    <td>${index + 1}</td>
+    <td>${element.name}</td>
+    <td>${element.price} &#8377;</td>
+    <td>${element.description}</td>
+    <td class="d-flex justify-content-center"><a href="/view.html?id=${index}" class="view-btn"><span class="material-symbols-outlined"> 
+    visibility
+    </span></a><div onclick="deleteData(${index})" class="del-btn"><span class="material-symbols-outlined">delete</span></div></td>
+    </tr>`;
+  });
+}
+// function to loads all data to our document
+showData();
+// Function to store data
+function addData() {
+  const pname = document.getElementById("name").value;
+  const description = document.getElementById("description").value;
+  const price = document.getElementById("price").value;
+  const image = document.getElementById("prodImg").src;
+  // if form is validated
+  if (validation() == true) {
+    let productdata;
+    if (localStorage.getItem("productdata") == null) {
+      productdata = [];
+    } else {
+      productdata = JSON.parse(localStorage.getItem("productdata"));
+    }
+    productdata.push({
+      name: pname,
+      price: price,
+      description: description,
+      image: image,
+    });
+    localStorage.setItem("productdata", JSON.stringify(productdata));
+    Swal.fire({
+      position: "center",
+      icon: "success",
+      title: "Data Added successfully",
+      showConfirmButton: true,
+    }).then((result) => {
+      if (result.isConfirmed) {
+        location.reload();
+      }
+    });
+    showData();
+    resetData();
+  }
+}
+
+// reset all entered data
+function resetData() {
+  document.getElementById("name").value = "";
+  document.getElementById("price").value = "";
+  document.getElementById("description").value = "";
+  document.getElementById("uploadFile").value = "";
+  newImg.style.display = "null";
+  newImg.setAttribute("src", "");
+}
+
+// deleting the data
+function deleteData(index) {
+  let productdata;
+  if (localStorage.getItem("productdata") == null) {
+    productdata = [];
+  } else {
+    productdata = JSON.parse(localStorage.getItem("productdata"));
+  }
+  Swal.fire({
+    title: "Are you sure?",
+    text: "You won't be able to revert this!",
+    icon: "warning",
+    showCancelButton: true,
+    confirmButtonColor: "#3085d6",
+    cancelButtonColor: "#d33",
+    confirmButtonText: "Yes, delete it!",
+  }).then((result) => {
+    if (result.isConfirmed) {
+      productdata.splice(index, 1);
+      localStorage.setItem("productdata", JSON.stringify(productdata));
+      showData();
+      resetData();
+      Swal.fire("Deleted!", "Your file has been deleted.", "success");
```

Comment:
Alert messages should not be misleading. Here we are deleting a record from the table.